lib/pos_blockchain.go

```diff
@@ -1851,22 +1832,81 @@ func (bc *Blockchain) GetCommittedTipView() (*UtxoView, error) {
 	return NewUtxoViewWithSnapshotCache(bc.db, bc.params, bc.postgres, bc.snapshot, nil, bc.snapshotCache)
 }
 
+// BlockViewAndUtxoOps is a struct that contains a UtxoView and the UtxoOperations
+// and a block that were used to build the UtxoView. This struct is only
+// used for Blockchain's blockViewCache, which is used to speed up repeated access
+// to a utxo view at an uncommitted block. Simply having a utxo view was insufficient
+// for all performance enhancements as the utxo operations are needed when committing
+// a block and the block is needed for the state syncer.
+type BlockViewAndUtxoOps struct {
+	UtxoView *UtxoView
+	UtxoOps  [][]*UtxoOperation
+	Block    *MsgDeSoBlock
+}
+
+func (viewAndUtxoOps *BlockViewAndUtxoOps) Copy() (*BlockViewAndUtxoOps, error) {
+	copiedView, err := viewAndUtxoOps.UtxoView.CopyUtxoView()
+	if err != nil {
+		return nil, errors.Wrapf(err, "BlockViewAndUtxoOps.Copy: Problem copying UtxoView")
+	}
+	return &BlockViewAndUtxoOps{
+		UtxoView: copiedView,
+		UtxoOps:  viewAndUtxoOps.UtxoOps,
```

Comment:
Would you not want to copy the UtxoOps too? Maybe there's a way to easily make them read-only so that some one can't modify them.